Added branch navigation shortcuts

Cankyre opened the pull request:
Added the navigation keybinds from #251 

![image](https://github.com/franciscoBSalgueiro/en-croissant/assets/75273490/90ee193b-4b47-4a23-8328-26cd0706020c)

NOTE: Default keybind to go to start/end (up/down) is remapped to branch start/end and is replaced with shift+up/down

Cankyre commented:
Forgot lint fix, sorry for the mess

franciscoBSalgueiro commented:
I tried to use "next branch" but it always gives me an error when I use it on the starting position. Maybe adding some tests in [store.test.ts](https://github.com/franciscoBSalgueiro/en-croissant/blob/master/src/utils/tests/store.test.ts) would help

TypeError: Cannot read properties of undefined (reading 'fen')
    at GameNotation (http://localhost:1420/src/components/boards/GameNotation.tsx:48:21)

Cankyre commented:
I fucked the pr, sorry, should be fixed though :/

Cankyre commented:
There, single commit

franciscoBSalgueiro commented:
I'm not convinced by the branch shortcuts. It doesn't feel as good as the lichess way of going to the next branch.

For example here:

1. e4 e5  (1... d6  2. d4  (2. d3  )) *

If I go to 1... d6 and then press ctrl-shiftleft it goes forward to 2. d4, which is very unintuitive. Lichess simply goes the next node where there's a branching.

Cankyre commented:
Indeed that is strange and not an expected behavior. In this case it should go to 1... e5. That seems to be a bug, I will investigate.

Cankyre commented:
If you always go to the next branching point, there won't be any way to navigate between the variations, making the keyboard navigation incomplete and require the mouse every now and then. A solution can be to have two different shortcuts or a single one that does both actions. I feel like going to the next branching is way less important than switching between branches

franciscoBSalgueiro commented:
I agree but having both in the same shortcut is a bit confusing. Maybe having tab for switching between branches when it's in a branching point would be better.

Cankyre commented:
I added two shortcuts for the branching point navigation (shift+right/left) and two for switching branch (x/c ; haven't found better)

P.S. I deleted the obsolete tests and have not took the time to make new ones yet, I can do that if necessary.

franciscoBSalgueiro commented:
Looks good! thank you
